App setup & Prayer times page

TarekkMA opened the pull request:
Still WIP PR.

TarekkMA committed:
remove pubspec.yaml comments

TarekkMA committed:
remove unused cupertino_icons dependency

cupertino_icons was causing the below issue.

The current Dart SDK version is 2.10.5.

Because cupertino_icons 1.0.2 requires SDK version >=2.12.0-0 <3.0.0 and no versions of cupertino_icons match >1.0.2 <2.0.0, cupertino_icons ^1.0.2 is forbidden.
So, because mcnd_mobile depends on cupertino_icons ^1.0.2, version solving failed.
pub get failed (1; So, because mcnd_mobile depends on cupertino_icons ^1.0.2, version solving failed.)

TarekkMA committed:
enable R8

will give better optimization to the android code.

TarekkMA committed:
upgrade packages

TarekkMA committed:
simple setup

TarekkMA committed:
delete tests for now

TarekkMA committed:
ignore generated files

TarekkMA committed:
add netowrking dependencies

TarekkMA committed:
app setup

TarekkMA committed:
add useOnce hook

TarekkMA committed:
add INTERNET permission to android

TarekkMA committed:
fix ext name

TarekkMA committed:
add hijri library

TarekkMA committed:
PrayerTimesWidget prototype

TarekkMA committed:
add bottom nav

TarekkMA committed:
using table

TarekkMA committed:
add header and sunrise

TarekkMA commented:
@gemy21ce I did complete the application setup and the general structure I think is now well-defined in my head,

I'm using the following packages:
 * `dio`: http client
 *  `enum_to_string` `strings`: useful utility functions
 * `hijri` convert to hijri date
 * `get_it` `injectable`: for dependency injection 
 * `riverpod`: dependency injection
 * `flutter_hooks` + `riverpod` + `state_notifier`: state management solution
 * `material_design_icons_flutter`: pretty icons
 * `pretty_dio_logger`: network logging
 * `freezed`: union/data classes
 * `json_serializable`: json serialization 


When you get the chance please review the UI, if every thing is okay I will refactor the UI code to make it cleaner and add some tests.

![Screen Shot 2021-03-30 at 10 12 10 PM](https://user-images.githubusercontent.com/6633545/113050377-0dacec80-91a5-11eb-81b9-792e3945382a.png)

gemy21ce commented on lib/data/models/app/salawat.dart:
```diff
@@ -0,0 +1,16 @@
+
+extension SalahExt on Salah {
+  String getStringName() {
+    return strings.capitalize(EnumToString.convertToString(this).toLowerCase());
```
why to lowercase then to capitalize?
and why not to use just this.toString().toUpperCase()

TarekkMA replied:
capitalize only make the first character upper case

TarekkMA committed:
rename salah file

TarekkMA committed:
move view logic to viewmodel

TarekkMA committed:
add datetime utils

+ format

TarekkMA committed:
move to utils package

TarekkMA committed:
extract duration to string logic

TarekkMA committed:
match salah time with a date

TarekkMA committed:
add debug current date

TarekkMA committed:
make nearestSalah visibleForTesting

TarekkMA committed:
mcnd_api_test.dart

+ changed api_prayer_time.dart to use freezed

TarekkMA committed:
datetime_utils_test.dart

TarekkMA committed:
fix: duration_utils.dart

TarekkMA committed:
test: duration_utils.dart

TarekkMA committed:
test: json_list_mapper.dart

TarekkMA committed:
rename symbols list_mapper.dart

TarekkMA committed:
test: prayer_time.dart mapper

TarekkMA committed:
move mapper to use mixins instead of extensions

TarekkMA committed:
test: prayer_times_viewmodel.dart

+ loading from api

TarekkMA committed:
app styles and colors adjustments

TarekkMA committed:
application icon

TarekkMA committed:
remove unused packages

TarekkMA committed:
remove redundant line

TarekkMA committed:
add WidgetsFlutterBinding.ensureInitialized() to main

TarekkMA committed:
add drawer and fix appbar

TarekkMA committed:
change border color to green

TarekkMA committed:
ui adjustments

TarekkMA committed:
update getTimeDifferenceString test

gemy21ce committed:
- model generator sh command

gemy21ce committed:
moving MCND logo from drawer to main prayers page.

TarekkMA committed:
move prayer_times_viewmodel_test.dart

TarekkMA committed:
Merge branch 'tarekkma-work' of https://github.com/gemy21ce/mcnd_mobile into tarekkma-work

TarekkMA committed:
move model_generator.sh to tools directory

TarekkMA committed:
Modify model_generator.sh

+ add shebang
+ calling with `model_generator.sh watch` will run in watch mode
+ added echo of what command is running

TarekkMA committed:
PrayerTimesPage widget tests

TarekkMA commented:
I think this PR is ready for a review

TarekkMA committed:
organize pubspec.yaml

TarekkMA committed:
fix tests

TarekkMA committed:
add a ci

TarekkMA committed:
+ add ios build to ci

TarekkMA committed:
run ci only for PR

TarekkMA commented:
I have added a CI check that does the following:
- run build_runner
- test
- build for android
- build for ios
